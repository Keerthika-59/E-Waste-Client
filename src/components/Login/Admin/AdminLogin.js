import React, { useState, useContext, useEffect } from "react";
import { Button, Image } from 'react-bootstrap';
import axios from 'axios';
import { BrowserRouter, Link, Redirect } from "react-router-dom";

import Cookies from 'js-cookie';
import AuthApi from '../../../authAPI';
import './style.css';
import Swal from 'sweetalert2'

const AdminLogin = (props) => {

    const Auth = useContext(AuthApi);
    const readCookies = () => {

        const admin = Cookies.get('admin');

        if (admin) {
            console.log('Admin Logged In');
            Auth.setAuth(true);
            props.history.push('/admin');
        }
    }

    useEffect(() => {
        readCookies();
    }, []);

    useEffect(() => {
        window.scrollTo(0, 0)
    }, [])

    const [email, setEmail] = useState();
    const [password, setPassword] = useState();
    const [show, setShow] = useState(false);

    const handleAdminLogin = async (e) => {

        e.preventDefault();

        try {

            if (email && password) {

                const login = await axios.post('https://ewaste-dec20-dev-api.azurewebsites.net/admin/login', {
                    email: email,
                    password: password
                });

                console.log(login);

                Auth.setAuth(true);
                Cookies.set("admin", login);
                props.history.push('/admin');
            }

            // await axios.post('')
        } catch (error) {

            Swal.fire('Login failed!','invalid credentials')

        }
    }
    return ((Cookies.get('admin')) ? <Redirect to='/admin' /> : (
        <div>
            <div className="container-fluid">
                <div className="row main-content bg-success text-center">
                    <div className="col-md-4 text-center company__info">
                        <span className="company__logo">
                            <Image
                                src='http://ewaste-dec20-dev.azurewebsites.net/assets/images/wm-logo.png'
                                width={70}
                                height={70}
                            />
                        </span>
                        <h4 className="company_title">Waste Management</h4>
                    </div>
                    <div className="col-md-8 col-xs-12 col-sm-12 login_form ">
                        <div className="container-fluid mx-auto">
                            <div className="row">

                                <form control="" className="form-group mx-auto" onSubmit={handleAdminLogin}>
                                    <div className="row">
                                        <input required={true} type="email" name="username" onChange={e => setEmail(e.target.value)} id="username" className="form__input" placeholder="Enter Your Email" />
                                    </div>

                                    <div className="row">
                                        <input required={true} type="password" onChange={e => setPassword(e.target.value)} name="password" id="password" className="form__input" placeholder="Enter Your Password" />
                                    </div>

                                    <div className="row">
                                        <Button disable={show} className="admin" type="submit" variant="info"> Log In </Button>
                                    </div>
                                </form>
                            </div>

                        </div>

                    </div>
                </div>
            </div>
        </div>)
    )
}

export default AdminLogin
