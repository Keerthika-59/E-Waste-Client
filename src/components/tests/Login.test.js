import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import UserForm from '../Login/userLogin';
import RepForm from '../Login/representativeLogin.js'
import AdminLogin from '../Login/Admin/AdminLogin'

test('renders test NavBar', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<UserForm/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });

   test('renders test NavBar', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<RepForm/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });

   test('renders test NavBar', () => {
    const div=document.createElement("div");
    <Route>
        ReactDom.render(<AdminLogin/>,div);
    </Route>
    ReactDOM.unmountComponentAtNode(div);
   });