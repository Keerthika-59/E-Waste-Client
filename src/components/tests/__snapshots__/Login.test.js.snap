// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AdminLogin component 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="row main-content bg-success text-center"
    >
      <div
        className="col-md-4 text-center company__info"
      >
        <span
          className="company__logo"
        >
          <Image
            fluid={false}
            height={70}
            rounded={false}
            roundedCircle={false}
            src="http://ewaste-dec20-dev.azurewebsites.net/assets/images/wm-logo.png"
            thumbnail={false}
            width={70}
          />
        </span>
        <h4
          className="company_title"
        >
          Waste Management
        </h4>
      </div>
      <div
        className="col-md-8 col-xs-12 col-sm-12 login_form "
      >
        <div
          className="container-fluid mx-auto"
        >
          <div
            className="row"
          >
            <form
              className="form-group mx-auto"
              control=""
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <input
                  className="form__input"
                  id="username"
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter Your Email"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="row"
              >
                <input
                  className="form__input"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter Your Password"
                  required={true}
                  type="password"
                />
              </div>
              <div
                className="row"
              >
                <Button
                  active={false}
                  className="admin"
                  disable={false}
                  disabled={false}
                  type="submit"
                  variant="info"
                >
                   Log In 
                </Button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should test RepForm component 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_excludedEdges": Array [],
      "_mutate": undefined,
      "_nodes": Array [
        "password",
        "email",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "email": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "email": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "password": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;

exports[`should test UserForm component 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_excludedEdges": Array [],
      "_mutate": undefined,
      "_nodes": Array [
        "password",
        "email",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "email": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "email": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "password": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
